ocrd_utils/ocrd_utils/logging.py

```diff
@@ -128,6 +135,17 @@ def initLogging():
     # To cut back on the `Self-intersection at or near point` INFO messages
     logging.getLogger('shapely.geos').setLevel(logging.ERROR)
 
+    if _overrideLogLevel:
+        setOverrideLogLevel(_overrideLogLevel)
```

Comment:
The sequence can be:

* setOverrideLoglevel (i.e. when using the `-l LEVEL` command line flag)
* initLogging

In this case, `_overrideLoglevel` will be set from `setOverrideLoglevel`. However, with logging uninitialized (beyond the basicConfig/CRITICAL) neither will this event have been logged nor are the loggers created that should be overridden. Hence the second call to setOverrideLoglevel after intialization.

Reply:
> However, with logging uninitialized (beyond the basicConfig/CRITICAL) neither will this event have been logged

I don't think so. Your `setOverrideLogLevel` already used the root loggers `.info()`, so at that point Python did run basicConfig and put the message to stdout. Afterwards `initLogging` destroys the handlers again.

> nor are the loggers created that should be overridden.

All loggers that have not been created already will still receive their `.setLevel(logging.NOTSET)` via `getLogger()`. And the root logger can still get `_overrideLogLevel` in `initLogging`'s `basicConfig`.

> Hence the second call to setOverrideLoglevel after intialization.

I don't see the need (see above). But if you want to keep it that way, I recommend at least naming the function differently to make clear it does not (just) set the global variable. How about differentiating `setOverrideLogLevel` and `applyOverrideLogLevel`?

Reply:
> Your setOverrideLogLevel already used the root loggers .info(), so at that point Python did run basicConfig and put the message to stdout.

That would work if we moved the "Overriding log level" info message after the `rootLogger.setLevel(lvl)` call, provided `lvl <= INFO`. Currently, this won't log at all because `logging.basicConfig(CRITICAL)`.

> All loggers that have not been created already will still receive their .setLevel(logging.NOTSET) via getLogger(). And the root logger can still get _overrideLogLevel in initLogging's basicConfig.

Agreed, that's a weak argument.

I'll test some more to see whether I can get rid of that second setOverrideLoglevel.

Reply:
The confusing second `setOverrideLoglevel` can be replaced by just assigning `_overrideLoglevel` to the root logger, if set, which is the only relevant function of setOverrideLoglevel in this situation. The overriding won't be logged on the command line but that was just noise to most users probably anyway.

No more `reinit` arg to `initLogging`.

No more exception in case of multiple initLogging but a critical log msg analogue to the warning about `getLogger` before `initLogging`